src/pymmcore_widgets/_pixel_size_widget.py

```diff
@@ -0,0 +1,590 @@
+import itertools
+import re
+import warnings
+from typing import Any, List, Optional, Tuple
+
+from pymmcore_plus import CMMCorePlus
+from qtpy import QtWidgets as QtW
+from qtpy.QtCore import QRegularExpression, Qt
+from qtpy.QtGui import QRegularExpressionValidator
+from superqt.utils import signals_blocked
+
+from ._objective_widget import ObjectivesWidget
+
+OBJECTIVE_LABEL = 0
+RESOLUTION_ID = 1
+CAMERA_PX_SIZE = 2
+MAGNIFICATION = 3
+IMAGE_PX_SIZE = 4
+
+FLOAT_REGEX = r"^\d*\.?\d*$"
+
+
+class PixelSizeTable(QtW.QTableWidget):
+    """Create a Table to set pixel size configurations."""
+
+    def __init__(self, parent: Optional[QtW.QWidget] = None) -> None:
+        super().__init__(parent)
+
+        hh = self.horizontalHeader()
+        hh.setSectionResizeMode(hh.Stretch)
+        hh.setDefaultAlignment(Qt.AlignHCenter)
+        vh = self.verticalHeader()
+        vh.setVisible(False)
+        vh.setSectionResizeMode(hh.Stretch)
+        self.setSelectionBehavior(QtW.QAbstractItemView.SelectItems)
+        self.setDragDropMode(QtW.QAbstractItemView.NoDragDrop)
+        self.setColumnCount(6)
+        self.setHorizontalHeaderLabels(
+            [
+                "Objective",
+                "ResolutionID",
+                "Camera Pixel Size (µm)",
+                "Magnification",
+                "Image Pixel Size (µm)",
+                "",
+            ]
+        )
+
+
+class PixelSizeWidget(QtW.QDialog):
+    """Make a widget to set the pixel size configuration.
+
+    Parameters
+    ----------
+    objective_device: str
+        A device label for which to create a widget.
+    parent : Optional[QWidget]
+        Optional parent widget.
+    """
+
+    def __init__(
+        self,
+        objective_device: str = "",
+        parent: Optional[QtW.QWidget] = None,
+        *,
+        mmcore: Optional[CMMCorePlus] = None,
+    ) -> None:
+        super().__init__(parent)
+
+        self.setWindowTitle("Set Image Pixel Size")
+
+        self._mmc = mmcore or CMMCorePlus.instance()
+
+        self._magnification: List[List[str, float]] = []  # type: ignore
```

Comment:
sometimes the type linter is telling you something you shouldn't ignore :joy:  `List[]` just doesn't take two arguments... if you want to specify the items in a list, you have to use `Tuple[str, float]`...

however... this _really_ kinda looks like it should just be a dictionary, (mapping objective label to magnification), no?

Reply:
(actually... it looks like this is unused altogether?? :joy:)